app/services/filtered_scope.rb

```diff
@@ -0,0 +1,42 @@
+class FilteredScope
+  def initialize(initial_scope, filters)
+    @initial_scope = initial_scope
+    @filters = filters
+  end
+
+  def scope
+    scope = initial_scope
+
+    [:section, :specialist_sector].each do |tag_type|
+      if filters[tag_type].present?
+        scope = scope.with_parent_tag(tag_type, filters[tag_type])
+      end
+    end
+
+    if filters[:state].present? && Artefact::STATES.include?(filters[:state])
+      scope = scope.in_state(filters[:state])
+    end
+
+    if filters[:kind].present? && Artefact::FORMATS.include?(filters[:kind])
+      scope = scope.of_kind(filters[:kind])
+    end
+
+    if filters[:search].present?
+      scope = scope.matching_query(filters[:search])
+    end
+
+    if filters[:owned_by].present?
+      scope = scope.owned_by(filters[:owned_by])
+    else
+      # Exclude all panopticon-owned artefacts from the index
+      # because they have their own specialised interfaces.
+      scope = scope.not_owned_by('panopticon')
+    end
+
+    scope
+  end
+
+private
+
+  attr_reader :initial_scope, :filters
```

Comment:
these getters are not private for real:

```ruby
irb(main):001:0> class FilteredScope
irb(main):002:1>    def initialize(initial_scope, filters)
irb(main):003:2>      @initial_scope = initial_scope
irb(main):004:2>      @filters = filters
irb(main):005:2>    end
irb(main):006:1>
irb(main):007:1*   attr_reader :initial_scope, :filters
irb(main):008:1>  end
```
`irb(main):009:0> a = FilteredScope.new(1,2)`
`irb(main):010:0> a.initial_scope`
=> 1
`irb(main):011:0> a.filters`
=> 2